src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexReadOperation.cs

```diff
@@ -184,13 +192,16 @@ private void SetupHighlighter(IndexQueryServerSide query, JsonOperationContext c
                     continue;
 
                 if (_tagsPerField == null)
-                    _tagsPerField = new Dictionary<string, (string[] PreTags, string[] PostTags)>();
+                    _tagsPerField = new();
 
                 var fieldName = query.Metadata.IsDynamic
                     ? throw new NotSupportedException("AutoIndex dynamic field is not supported yet.")
                     : highlighting.Field.Value;
+                
+                throw new NotImplementedException();
```

Comment:
Can anyone give me a hint on how on earth can I get the actual terms I am looking for to highlight later? I understand the complexity of the `Query` but it seems to me that we are missing a few levels of separation from Lucene (or I am not understanding the abstraction). 

cc @maciejaszyk

Reply:
Let's talk about that directly.

Reply:
https://github.com/maciejaszyk/ravendb/tree/RavenDB-17599.terms @redknightlois